### PR TITLE
More LSP compatibility on arg names

### PR DESCRIPTION
**Pull request by hauntsaninja**
Got lost when #18356 was broken up

### CODE REVIEW

File: mypy/plugin.py [hunk 1]
```diff
@@ -170,12 +170,12 @@ def fail(self, msg: str, ctx: Context, *, code: ErrorCode | None = None) -> None
         raise NotImplementedError
 
     @abstractmethod
-    def named_type(self, name: str, args: list[Type]) -> Instance:
+    def named_type(self, fullname: str, args: list[Type], /) -> Instance:
```
Comment (cdce8p) [inline mypy/plugin.py:173]: Changing the argument to `fullname` does make sense. Just not sure about adding `/` though. It feels a bit like an escape hatch. For example it hides the fact that the argument is still named something different in the implementation. I'd suggest removing the `/` here.

https://github.com/python/mypy/blob/ac6151ae6657a8bbd9793c74691c9985dae02c15/mypy/typeanal.py#L1939-L1941

File: mypy/plugin.py [hunk 2]
```diff
@@ -170,12 +170,12 @@ def fail(self, msg: str, ctx: Context, *, code: ErrorCode | None = None) -> None
...
         """Construct an instance of a builtin type with given name."""
         raise NotImplementedError
 
     @abstractmethod
-    def analyze_type(self, typ: Type) -> Type:
+    def analyze_type(self, typ: Type, /) -> Type:
```
Comment (cdce8p) [inline mypy/plugin.py:178]: Same case here. The `/` feels unnecessary. Changing the argument name to `typ` in the single implementation would probably be better.
https://github.com/python/mypy/blob/ac6151ae6657a8bbd9793c74691c9985dae02c15/mypy/typeanal.py#L1765-L1766

File: mypy/plugin.py [hunk 3]
```diff
@@ -319,7 +319,8 @@ def fail(
     @abstractmethod
     def anal_type(
         self,
-        t: Type,
+        typ: Type,
+        /,
```
Comment (cdce8p) [inline mypy/plugin.py:323]:
```suggestion
        typ: Type,
```

File: mypy/semanal_shared.py [hunk 4]
```diff
@@ -176,7 +176,8 @@ def accept(self, node: Node) -> None:
     @abstractmethod
     def anal_type(
         self,
-        t: Type,
+        typ: Type,
+        /,
```
Comment (cdce8p) [inline mypy/semanal_shared.py:180]:
```suggestion
        typ: Type,
```

File: mypy/plugin.py [hunk 5]
```diff
@@ -340,15 +341,15 @@ def class_type(self, self_type: Type) -> Type:
         raise NotImplementedError
 
     @abstractmethod
-    def lookup_fully_qualified(self, name: str) -> SymbolTableNode:
+    def lookup_fully_qualified(self, fullname: str, /) -> SymbolTableNode:
```
Comment (cdce8p) [inline mypy/plugin.py:344]:
```suggestion
    def lookup_fully_qualified(self, fullname: str) -> SymbolTableNode:
```
I'd again update the implementation instead.
https://github.com/python/mypy/blob/ac6151ae6657a8bbd9793c74691c9985dae02c15/mypy/typeanal.py#L291-L292

File: mypy/semanal_shared.py [hunk 6]
```diff
@@ -76,11 +76,11 @@ def lookup_qualified(
         raise NotImplementedError
 
     @abstractmethod
-    def lookup_fully_qualified(self, name: str) -> SymbolTableNode:
+    def lookup_fully_qualified(self, fullname: str, /) -> SymbolTableNode:
```
Comment (cdce8p) [inline mypy/semanal_shared.py:79]:
```suggestion
    def lookup_fully_qualified(self, fullname: str) -> SymbolTableNode:
```

File: mypy/plugin.py [hunk 7]
```diff
@@ -340,15 +341,15 @@ def class_type(self, self_type: Type) -> Type:
...
         """Lookup a symbol by its fully qualified name.
 
         Raise an error if not found.
         """
         raise NotImplementedError
 
     @abstractmethod
-    def lookup_fully_qualified_or_none(self, name: str) -> SymbolTableNode | None:
+    def lookup_fully_qualified_or_none(self, fullname: str, /) -> SymbolTableNode | None:
```
Comment (cdce8p) [inline mypy/plugin.py:352]:
```suggestion
    def lookup_fully_qualified_or_none(self, fullname: str) -> SymbolTableNode | None:
```

File: mypy/semanal_shared.py [hunk 8]
```diff
@@ -76,11 +76,11 @@ def lookup_qualified(
...
         raise NotImplementedError
 
     @abstractmethod
-    def lookup_fully_qualified_or_none(self, name: str) -> SymbolTableNode | None:
+    def lookup_fully_qualified_or_none(self, fullname: str, /) -> SymbolTableNode | None:
```
Comment (cdce8p) [inline mypy/semanal_shared.py:83]:
```suggestion
    def lookup_fully_qualified_or_none(self, fullname: str) -> SymbolTableNode | None:
```

**Review by cdce8p — CHANGES_REQUESTED** [review]

**Review by hauntsaninja — COMMENTED** [review]
I prefer positional-only arguments here, since this is part of mypy's plugin API. This way plugin authors get errors upfront, instead of perhaps later when running their plugin against an older version of mypy.

I can make some of the implementations positional-only too though.

**Comment by cdce8p** [comment]
> I prefer positional-only arguments here, since this is part of mypy's plugin API. This way plugin authors get errors upfront, instead of perhaps later when running their plugin against an older version of mypy.

Makes sense. The methods marked with pos-only are unlikely to be called with keyword args anyway. Would you mind updating the argument names in the implementation, so that these are at least consistent (i.e. the ones I liked to above)?

**Review by cdce8p — APPROVED** [review]
Thanks @hauntsaninja 👍🏻 
Last thing, could you also update the comment on the API changes issue? _I don't have the permission to do that, unfortunately._

**Pull request merged**